pkg/virt-handler/non-root.go

```diff
@@ -106,29 +107,30 @@ func (d *VirtualMachineController) prepareStorage(vmi *v1.VirtualMachineInstance
 	return changeOwnershipOfHostDisks(vmi, res)
 }
 
-func getTapDevices(vmi *v1.VirtualMachineInstance) (map[string]string, error) {
-	macvtap := map[string]struct{}{}
+func getTapDevices(vmi *v1.VirtualMachineInstance, networkBindings map[string]v1.InterfaceBindingPlugin) (map[string]string, error) {
+	tapNetworks := map[string]struct{}{}
+	domainAttachmentByInterfaceName := domainspec.DomainAttachmentByInterfaceName(vmi.Spec.Domain.Devices.Interfaces, networkBindings)
 	for _, inf := range vmi.Spec.Domain.Devices.Interfaces {
-		if inf.Macvtap != nil {
-			macvtap[inf.Name] = struct{}{}
+		if domainAttachmentByInterfaceName[inf.Name] == string(v1.Tap) && inf.Masquerade == nil && inf.Bridge == nil {
```

Comment:
We can generalize it and everything will perfectly work if `inf.Masquerade == nil && inf.Bridge == nil` will be removed from the condition.
The reason I prefer to exclude it is because the next step is to step into the virt-laucnher's network namespace and looking for the device. It is a bit expensive, so if we can avoid it. Why not?
Another bindings don't have this privilege (that we know in advance if the tap device is there or not), so they we have to step into the network namespace to check.